### PR TITLE
Dependencies cleanup to allow python 3.8

### LINKED ISSUE #372: numpy no longer needs to be pinned to 1.20
### Information

- **Qiskit Metal version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?
numpy is pinned to 1.20 due to Qutip issues back in mid-February.
See https://github.com/qutip/qutip/issues/1442

### Steps to reproduce the problem
See link above

### What is the expected behavior?
See link above

### LINKED ISSUE #514: Enable higher python versions, all OSes, and pip venv
Observations on Windows:

Removing python limitation installs python 3.9 but leads to a shiboken2 error when importing pyside2.
Observations: the pip install of qiskit-metal overwrites the conda pyside2 install
- conda has pyside2 5.13 (max version for all python versions)
- PyPL has only pyside2 5.15.2 for python 3.9 (5.14 for 3.8 and 5.13 for 3.7)

Tests:
- Removing pyside2 from requirements.txt keeps the 5.13 conda version, which works well, thus the qiskit-metal installation leads to successful execution
- Doing the opposite (removing pyside2 dependency from environment.yml) leads to a jupyter notebook kernel crash during creation of a design
- however we need pyside2 in both requirements.txt and environment.yml to be able to deploy qiskit-metal to PyPI, so we need to find a way to make things work for both conda and PyPI deploys on a variety of python versions.

Todo:
- More tests to find optimal installation flows for all three cases: conda, PyPI and github clone, for a variety of python versions.

### PR DESCRIPTION
**Pull request by marcolincs**
### What are the issues this pull addresses (issue numbers / links)?
Closes #372
Covers partly #514
Blocked by approval of #512

### Did you add tests to cover your changes (yes/no)?
no
### Did you update the documentation accordingly (yes/no)?
no
### Did you read the CONTRIBUTING document (yes/no)?
yes
### Summary

Thanks to #512, we remove the major factor that imposed dependency limit.
With this PR we try to relax the dependency versions as much as possible.

Reason of outstanding python version upper limit:
-> conda Windows OS: install without upper limit, leads to a shiboken2 error. The error is caused by the pip version of PySide2 (5.15.2). In contrast, the conda version of PySide2 (5.13.2) works just fine on python 3.9, thus no shiboken2 error occurs. Unfortunately, the conda version is overwritten by the pip version during the pip install of qiskit-metal. We cannot remove pyside2 from requirements.txt because we need it there for the PyPI deploy.
It was noticed that pip PySide2 version 5.15.0 works correctly, but 5.15.0 requires python 3.8, thus the python version upper limit is currently set to <3.9, unless solution is found. Will continue this investigation in #514

### CODE REVIEW

**Comment by marcolincs** [comment]
@grace-harper-ibm , can you please test this new environment.yml and requirements.txt on mac? I will test it on linux

**Comment by marcolincs** [comment]
Tested on Linux: the README.md installation flow works fine with these relaxed requirements.

**Comment by marcolincs** [comment]
I updated the github actions, and tested the MAC that way. looks clean.
Still cannot include windows in the testing because the pip install of geopandas in windows requires the fiona libraries, which do not exist in PyPI for windows. Will address this issue as a follow-up, if possible.

**Review by JeremyDrysdale-ibm — APPROVED** [review]
LGTM

**Comment by grace-harper-ibm** [comment]
If I edit `run_all_tests` to run without `pytest` (we've removed `pytest` previously), I've found the conda environment I created from this PR causes errors with the drawing packages. 
Example:
```
Test two_points_described in Vector class in utility.py. ... ok
======================================================================
FAIL: test_draw_basic_subtract (__main__.TestDraw)
Test subtract in basic.py.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_draw.py", line 90, in test_draw_basic_subtract
    self.assertEqual(actual_subtract[i][j], expected_subtract[i][j])
AssertionError: 3.95 != 2.85
======================================================================
FAIL: test_draw_basic_union (__main__.TestDraw)
Test union in basic.py.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_draw.py", line 115, in test_draw_basic_union
    self.assertAlmostEqualRel(actual[x][i][j],
  File "/Users/graceharperibm/metal_work/qiskit-metal/qiskit_metal/tests/assertions.py", line 62, in assertAlmostEqualRel
    raise self.failureException(msg)
AssertionError: Relative difference 0.27848101265822783 exceeds tolerance 0.001 for expected=3.95, tested=2.85
----------------------------------------------------------------------
Ran 31 tests in 0.009s
FAILED 
```

**Comment by JeremyDrysdale-ibm** [comment]
> If I edit `run_all_tests` to run without `pytest` (we've removed `pytest` previously), I've found the conda environment I created from this PR causes errors with the drawing packages.
> Example:
> 
> ```
> Test two_points_described in Vector class in utility.py. ... ok
> ======================================================================
> FAIL: test_draw_basic_subtract (__main__.TestDraw)
> Test subtract in basic.py.
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "test_draw.py", line 90, in test_draw_basic_subtract
>     self.assertEqual(actual_subtract[i][j], expected_subtract[i][j])
> AssertionError: 3.95 != 2.85
> ======================================================================
> FAIL: test_draw_basic_union (__main__.TestDraw)
> Test union in basic.py.
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "test_draw.py", line 115, in test_draw_basic_union
>     self.assertAlmostEqualRel(actual[x][i][j],
>   File "/Users/graceharperibm/metal_work/qiskit-metal/qiskit_metal/tests/assertions.py", line 62, in assertAlmostEqualRel
>     raise self.failureException(msg)
> AssertionError: Relative difference 0.27848101265822783 exceeds tolerance 0.001 for expected=3.95, tested=2.85
> ----------------------------------------------------------------------
> Ran 31 tests in 0.009s
> FAILED 
> ```
> 
> **These error messages likely don't show up in our CI because they seem to be an issue with the conda env created and our CI doesn't use conda. So all green checks appear to hide this error**

Strange... I didn't have those failures in a conda environment.

**Comment by marcolincs** [comment]
The test error was just that, a test error. I updated the test to account for possible shift of the points after the union and subtraction operation. Posted updates.

Aside from this fixed issue, I believe @grace-harper-ibm also experienced WARNINGS/CRITICAL flags when trying to run the jupyter notebook deep dive. I remember seeing the warnings before during my tests but they only showed at random and I cannot reproduce them now. I have never seen the CRITICAL flags. Will need Grace (or other mac user) to debug those.

Suggested approach:
- making sure those only occur in the branch and not in main
- grab a `conda list` from the two conda envs (main and branch)
- compare the two lists for differences.
- gradually upgrade the libraries in the conda environment created with main.
NOTE: remember to install qiskit-metal as `python -m pip install --no-deps -e .` to avoid pip libraries to get in the mix and add confusion.

**Comment by ssilverman99-coder** [comment]
I am having issues with import or RouteMeander.  It works on several other
machines I use .
[image: image.png]

**Comment by marcolincs** [comment]
@ssilverman99-coder , I am somewhat confused.
1) Is this related to this draft PR? if yes, please see point 2. If not, please create a new issue and for this and delete your comment in this thread.
2) there is no information to understand your post. The image did not go through, but even if it did, that would not suffice to help you, you would need to post how you got there.

Regards,

**Review by JeremyDrysdale-ibm — APPROVED** [review]

**Review by grace-harper-ibm — COMMENTED** [review]
There is a critical error showing up in the Jupyter Noteboks on Metal's launch:
```
12:40PM 40s CRITICAL [_qt_message_handler]: line: 0, func: None(), file: None  WARNING: Layer-backing can not be explicitly controlled on 10.14 when built against the 10.14 SDK

12:40PM 40s CRITICAL [_qt_message_handler]: line: 0, func: None(), file: None  CRITICAL: <QNSWindow: 0x7fb822f4afb0; contentView=<QNSView: 0x7fb822f4a8a0; QCocoaWindow(0x7fb822f4a780, window=QWidgetWindow(0x7fb822f4a2e0, name="MainWindowPlotWindow"))>> has active key-value observers (KVO)! These will stop working now that the window is recreated, and will result in exceptions when the observers are removed. Break in QCocoaWindow::recreateWindowIfNeeded to debug.
```


However, met with Marco and it seems the critical error is on Mac only and does not appear to affect the look nor functionality of Metal. 

Marco says to add this knowledge to the FAQ so our customers are not confused

**Comment by ssilverman99-coder** [comment]
Marco--Sorry to bug ya with the issue.  I was in the "wrong" environment.
All is working OK now!  Also are there any sample ANYSYS Electronics HFSS
examples to play with.

Thanks

File: qiskit_metal/_defaults.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def update_default_options(self,
                                cust_value: Dict = None):
         """Allow instance of class to update the default_options.
 
-        Args:
+        Arguments:
```
Comment (marcolincs) [inline qiskit_metal/_defaults.py:130]: Hi @grace-harper-ibm ,
I am somewhat confused on this change here. This is inconsistent with the VScode default, which is `Args:`. Making this `Arguments:` will constantly cause inconsistency in future PR that use the default. Any particular reason you went ahead with this change (in this PR)?

**Comment by marcolincs** [comment]
> Marco--Sorry to bug ya with the issue. I was in the "wrong" environment. All is working OK now! Also are there any sample ANYSYS Electronics HFSS examples to play with. Thanks

@ssilverman99-coder , you also still are in the wrong issue. Please either create a new issue, or reach out on Slack. Thanks

File: qiskit_metal/_defaults.py (hunk 1 above)
Comment (JeremyDrysdale-ibm) [reply qiskit_metal/_defaults.py:130]: `Args:` and `Arguments:` are both acceptable from a docs standpoint, Presently, they are used interchangeably throughout the code.

Comment (JeremyDrysdale-ibm) [reply qiskit_metal/_defaults.py:130]: That having been said, changing `Args:` to `Arguments` in this case is unnecessary.

Comment (marcolincs) [reply qiskit_metal/_defaults.py:130]: I agree they are both acceptable from a docstring stand point, but given that out IDE of choice uses `Args:` by default, changing things the opposite way feels like asking for trouble.

Comment (grace-harper-ibm) [reply qiskit_metal/_defaults.py:130]: Hey Marco
That's a good point. I just saw things were inconsistent and so I arbitrarily chose one to standardize across the board
Let me remedy it

**Review by marcolincs — COMMENTED** [review]
I am good with the Arguments->Args change. Not too noisy on this CR. Did a couple of reverts

**Review by JeremyDrysdale-ibm — APPROVED** [review]
LGTM

**Pull request merged**